src/relay.cpp

```diff
@@ -785,63 +794,71 @@ void dhcp6relay_stop()
  * 
  * @brief               main loop: configure sockets, create libevent base, start server listener thread
  *  
  * @param vlans         list of vlans retrieved from config_db
  * @param db            state_db connector
  */
-void loop_relay(std::vector<relay_config> *vlans, swss::DBConnector *db) {
+void loop_relay(std::vector<relay_config> *vlans) {
     std::vector<int> sockets;
     base = event_base_new();
     if(base == NULL) {
         syslog(LOG_ERR, "libevent: Failed to create base\n");
     }
 
+    std::shared_ptr<swss::DBConnector> state_db = std::make_shared<swss::DBConnector> ("STATE_DB", 0);
+    std::shared_ptr<swss::DBConnector> config_db = std::make_shared<swss::DBConnector> ("CONFIG_DB", 0);
+    std::shared_ptr<swss::Table> mStateDbMuxTablePtr = std::make_shared<swss::Table> (
+            state_db.get(), "HW_MUX_CABLE_TABLE"
+        );
+    db.config_db = config_db;
+    db.muxTable = mStateDbMuxTablePtr;
+
+    int filter = 0;
+    filter = sock_open(&ether_relay_fprog);
+    sockets.push_back(filter);
+
     for(relay_config &vlan : *vlans) {
         relay_config *config = &vlan;
-        int filter = 0;
         int local_sock = 0; 
         int server_sock = 0;
-        int index = if_nametoindex(config->interface.c_str());
-        config->db = db;
+        config->state_db = state_db;
+        config->mux_key = vlan_member + config->interface + "|";
 
         std::string counterVlan = counter_table;
-        initialize_counter(config->db, counterVlan.append(config->interface));
-
-        filter = sock_open(index, &ether_relay_fprog);
-        prepare_socket(&local_sock, &server_sock, config, index);
+        initialize_counter(config->state_db, counterVlan.append(config->interface));
+        prepare_socket(&local_sock, &server_sock, config);
 
         config->local_sock = local_sock;
         config->server_sock = server_sock;
 
-        sockets.push_back(filter);
         sockets.push_back(local_sock);
         sockets.push_back(server_sock);
-
         prepare_relay_config(config, &local_sock, filter);
 
         evutil_make_listen_socket_reuseable(filter);
         evutil_make_socket_nonblocking(filter);
 
         evutil_make_listen_socket_reuseable(local_sock);
         evutil_make_socket_nonblocking(local_sock);
-
+    
         listen_event = event_new(base, filter, EV_READ|EV_PERSIST, callback, config);
         server_listen_event = event_new(base, local_sock, EV_READ|EV_PERSIST, server_callback, config);
+
         if (listen_event == NULL || server_listen_event == NULL) {
             syslog(LOG_ERR, "libevent: Failed to create libevent\n");
         }
 
         event_add(listen_event, NULL);
         event_add(server_listen_event, NULL);
-    }
-    
-    if((signal_init() == 0) && signal_start() == 0) {
-        shutdown();
-        for(std::size_t i = 0; i<sockets.size(); i++) {
-            close(sockets.at(i));
+        
+        if((signal_init() == 0) && signal_start() == 0) {
```

Comment:
Should this be inside a for-loop? I think this should only be run once.

Reply:
Moved it outside of for loop